pkg/ctl/gitops/apply.go

```diff
@@ -0,0 +1,178 @@
+package gitops
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/client-go/tools/clientcmd"
+
+	kubeclient "k8s.io/client-go/kubernetes"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+	"github.com/weaveworks/eksctl/pkg/git"
+	"github.com/weaveworks/eksctl/pkg/gitops"
+	"github.com/weaveworks/eksctl/pkg/gitops/fileprocessor"
+	"github.com/weaveworks/eksctl/pkg/gitops/flux"
+)
+
+const (
+	defaultGitTimeout = 20 * time.Second
+)
+
+type options struct {
+	git               git.Options
+	quickstartNameArg string
+	outputPath        string
+}
+
+func applyGitops(cmd *cmdutils.Cmd) {
+	cfg := api.NewClusterConfig()
+	cmd.ClusterConfig = cfg
+
+	cmd.SetDescription("apply", "Setting up GitOps and apply a Quick Start profile", "")
+
+	var opts options
+
+	cmd.SetRunFuncWithNameArg(func() error {
+		return doApplyGitops(cmd, opts)
+	})
+
+	cmd.FlagSetGroup.InFlagSet("General", func(fs *pflag.FlagSet) {
+		fs.StringVarP(&opts.quickstartNameArg, "quickstart-profile", "", "", "name or URL of the Quick Start profile. For example, app-dev.")
+		fs.StringVarP(&opts.git.URL, "git-url", "", "", "URL for the git repository that will contain the cluster components")
```

Comment:
Nit: I think we're starting to have several commands which accept Git-related arguments.
It may be worth factorising these -- to help reduce boilerplate and enforce consistency across commands -- in, for example, a `AddCommonFlagsForGit` function in `pkg/ctl/cmdutils/cmdutils.go`?

Reply:
I think I would need to sit for a bit and reorganize that. We can address that in another Pr.